veniq/dataset_collection/augmentation.py

```diff
@@ -136,55 +122,125 @@ def _is_match_to_the_conditions(
 
-def determine_type(method_node: ASTNode):
+def check_whether_method_has_return_type(
+        method_decl: AST,
+        var_decls: typing.Set[str]) -> InlineTypesAlgorithms:
     """
-
-    :param method_node: Method declaration
-    :return: int - type
+    Run function to check whether Method declaration can be inlined
+    :param method_decl: method, where invocation occurred
+    :param var_decls: set of variables for found invoked method
+    :return: enum InlineTypesAlgorithms
     """
+    names = get_variables_decl_in_node(method_decl)
 
-    if not method_node.parameters:
-        return 0
-    else:
-        return -1
+    var_decls_original = set(names)
+    intersected_names = var_decls & var_decls_original
+    # if we do not have intersected name in target method and inlined method
+    # and if we do not have var declarations at all
+    if not var_decls or not intersected_names:
+        return InlineTypesAlgorithms.WITHOUT_RETURN_WITHOUT_ARGUMENTS
+
+    return InlineTypesAlgorithms.DO_NOTHING
+
+
+def get_variables_decl_in_node(
+        method_decl: AST) -> List[str]:
+    names = []
+    for x in method_decl.get_proxy_nodes(ASTNodeType.VARIABLE_DECLARATOR):
+        if hasattr(x, 'name'):
+            names.append(x.name)
+        elif hasattr(x, 'names'):
+            names.extend(x.names)
+
+    for x in method_decl.get_proxy_nodes(ASTNodeType.VARIABLE_DECLARATION):
+        if hasattr(x, 'name'):
+            names.append(x.name)
+        elif hasattr(x, 'names'):
+            names.extend(x.names)
 
+    for x in method_decl.get_proxy_nodes(ASTNodeType.TRY_RESOURCE):
+        names.append(x.name)
 
-def _create_new_files(
+    return names
+
+
+def determine_algorithm_insertion_type(
+        ast: AST,
+        method_node: ASTNode,
+        invocation_node: ASTNode,
+        dict_original_nodes: Dict[str, List[ASTNode]]
+) -> InlineTypesAlgorithms:
+    """
+
+    :param ast: ast tree
+    :param dict_original_nodes: dict with names of function as key
+    and list of ASTNode as values
+    :param method_node: Method declaration. In this method invocation occurred
+    :param invocation_node: invocation node
+    :return: InlineTypesAlgorithms enum
+    """
+
+    original_invoked_method = dict_original_nodes.get(invocation_node.member, [])
+    # ignore overridden functions
+    if (len(original_invoked_method) == 0) or (len(original_invoked_method) > 1):
+        return InlineTypesAlgorithms.DO_NOTHING
+    else:
+        original_method = original_invoked_method[0]
+        if not original_method.parameters:
+            if not original_method.return_type:
+                # Find the original method declaration by the name of method invocation
+                var_decls = set(get_variables_decl_in_node(ast.get_subtree(original_method)))
+                return check_whether_method_has_return_type(
+                    ast.get_subtree(method_node),
+                    var_decls
+                )
+            else:
```

Comment:
A lot of else statements. Kind of difficult to read. Can you refactor it a bit with less indentation depth.